plugin/trino-memory/src/main/java/io/trino/plugin/memory/MemoryMetadata.java

```diff
@@ -417,4 +418,13 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
 
         return Optional.of(new MemoryTableHandle(table.getId(), table.getLimit(), OptionalDouble.of(table.getSampleRatio().orElse(1) * sampleRatio)));
     }
+
+    @Override
+    public String canonicalize(ConnectorSession session, String identifier, boolean delimited)
+    {
+        if (delimited) {
+            return identifier;
+        }
+        return identifier.toUpperCase(ENGLISH);
```

Comment:
That could be a breaking change. Memory connector is used. Maybe you could use `MockConnector` for experiments like that?

Reply:
Or can we have it in a legacy way ?

Reply:
No. Memory connector is a dead end. It is limited while we need to test most of features. Using `MockConnector` as can be used for everything (it might just need some plumbing). `MockConnector` can be used to inject any functionality you may imagine so it is better suited for such testing.